core/modules/filters/mediaquery.js

```diff
@@ -0,0 +1,97 @@
+/*\
+title: $:/core/modules/filters/mediaquery.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator for evaluating CSS media queries
+
+\*/
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+// Store active media query listeners
+var mediaQueryListeners = {};
+var listenerCounter = 0;
+
+/*
+Export our filter function
+*/
+exports.mediaquery = function(source,operator,options) {
+	var results = [],
+		mediaQuery = operator.operand,
+		widget = options.widget;
+	
+	// Only evaluate in browser environment
+	if($tw.browser && mediaQuery && widget) {
+		try {
+			// Use window.matchMedia to evaluate the media query
+			var mql = window.matchMedia(mediaQuery);
+			
+			// Process each input title
+			source(function(tiddler,title) {
+				if(mql.matches) {
+					results.push(title);
+				}
+			});
+			
+			// Set up a listener for changes if we have a widget context
+			if(widget && widget.wiki) {
+				// Create a unique key for this listener
+				var listenerKey = "mql_" + (++listenerCounter);
+				
+				// Define the change handler
+				var changeHandler = function(e) {
+					// Force a refresh of the widget when media query changes
+					if(widget.refreshSelf) {
+						widget.refreshSelf();
+					} else if(widget.parentWidget && widget.parentWidget.refreshChildren) {
+						widget.parentWidget.refreshChildren();
+					}
+				};
+				
+				// Add the listener (use modern addEventListener if available)
+				if(mql.addEventListener) {
+					mql.addEventListener("change", changeHandler);
```

Comment:
I just saw this. I am not sure, if you can add an event-handler to a filter operator
@Jermolene - Is that possible?

Reply:
It's possible in terms of `it's doable` 😁 but if it's allowed in the core - that's a different question.
But this event listener would make this filter operator `special` because it doesn't react to tiddler changes but to media-query changes